packages/internal-test-env/index.ts

```diff
@@ -78,100 +77,97 @@ export interface TestingEnvironmentBase {
     );
   }
-
-  // Creating feature flag object if there are feature flags
-  Object.keys(process.env)
-    .filter((envVar) => envVar.startsWith("E2E_TEST_FEATURE_"))
-    .forEach((key) => (featuredFlags[key] = process.env[key]));
-  // Marking env as loaded
   envLoaded = true;
 }
 
-function getTestingEnvironment(
-  environment: unknown
-): asserts environment is EnvVariables {
-  // Populate your process.env from your .env file of choice
+function getBaseTestingEnvironment<T extends LibraryVariables>(
+  libVars?: T
+): T extends NodeVariables
+  ? TestingEnvironmentNode
+  : TestingEnvironmentBrowser {
+  setupEnv();
+
+  // Load and validate target environment name.
+  const targetEnvName = process.env.E2E_TEST_ENVIRONMENT;
+  if (!availableEnvironments.includes(targetEnvName as AvailableEnvironments)) {
+    throw new Error(`Unknown environment: [${targetEnvName}]`);
+  }
 
-  // TODO: Replace these inline validations and checks with envalid or env-var
-  if (
-    !availableEnvironment.includes(
-      (environment as EnvVariables).E2E_TEST_ENVIRONMENT as AvailableEnvironment
-    )
-  ) {
+  // Load and validate target OpenID Provider.
+  const targetOp = process.env.E2E_TEST_IDP;
+  if (typeof targetOp !== "string" || !isValidUrl(targetOp)) {
     throw new Error(
-      `Unknown environment: [${
-        (environment as EnvVariables).E2E_TEST_ENVIRONMENT
-      }]`
+      `The environment variable E2E_TEST_IDP is not a valid URL: found ${targetOp}.`
     );
   }
 
-  if (typeof (environment as EnvVariables).E2E_TEST_IDP !== "string") {
-    throw new Error("The environment variable E2E_TEST_IDP is undefined.");
-  }
-}
+  // Creating feature flag object if there are feature flags
+  const features = Object.keys(process.env)
+    .filter((envVar) => envVar.startsWith(ENV_VAR_PREFIX))
+    .reduce(
+      (featureFlags, envVar) => ({
+        ...featureFlags,
+        [`${envVar}`]: process.env[envVar]?.split(ENV_VAR_PREFIX)[1],
+      }),
+      {}
+    );
 
-function getBaseTestingEnvironment<T extends LibraryVariables>(libVars?: T
-  ): T extends NodeVariables ? TestingEnvironmentNode : TestingEnvironmentBrowser   {
-    setupEnv();
-    getTestingEnvironment(process.env);
-  
-    const base = {
-      idp: process.env.E2E_TEST_IDP,
-      environment: process.env.E2E_TEST_ENVIRONMENT,
-      features: featuredFlags,
-    };
-  
-    return libVars ? merge(base, validateLibVars(libVars)) : base;
+  const base = {
+    idp: targetOp,
+    environment: targetEnvName,
+    features,
+  };
+
+  return libVars ? merge(base, validateLibVars(libVars)) : base;
 }
 
 export function getNodeTestingEnvironment(
   varsToValidate?: LibraryVariables
 ): TestingEnvironmentNode {
-  return getBaseTestingEnvironment(merge(varsToValidate, {
-    // Enforce client credentials are present for the resource owner.
-    clientCredentials: { owner: { id: true, secret: true }}
-  }) as NodeVariables);
+  return getBaseTestingEnvironment(
```

Comment:
Perhaps `getBaseTestingEnvironment<NodeVariables>(` instead? That way you don't need the `as NodeVariables` afaict.

Reply:
https://github.com/inrupt/typescript-sdk-tools/pull/113/commits/5493567377bb3b9cf8b2652473911fa2329339f8